### PR TITLE
Timestamps should use currentTimeMillis 

### PR DESCRIPTION
**Pull request by tgodzik**
`nanoTime` is not actually time in nanos from 1970, but is used to measure difference. So all timestamps would be quite random since they can repeat themselves. We can't use them as timestamps for sure.

Also add pom.xml to be watched, since I must have forgot about it previously.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/Time.scala [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
-import java.util.concurrent.TimeUnit
-
 /**
  * Wrapper around `System.nanoTime` to allow easier testing.
  */
 trait Time {
   def nanos(): Long
-  final def millis(): Long = TimeUnit.NANOSECONDS.toMillis(nanos())
+  final def millis(): Long = System.currentTimeMillis()
```
Comment (gabro) [inline metals/src/main/scala/scala/meta/internal/metals/Time.scala]: Beware: this is not related to the nanos method anymore, which I think it was the intention

Comment (ivanopagano) [reply metals/src/main/scala/scala/meta/internal/metals/Time.scala]: You should check how the `Time` trait is used for testing, since that seems to be the original intent.
As @gabro points out, the idea might be that `millis()` is coupled to `nanos()` for testing purposes, and the `Time` is just a commodity util over nanotimes...

If that's the case, I'd suggest renaming `millis()` to something more explicit like `asMillis()` and document that.
Or even better, create an additional `millisTimestamp()` that doesn't relates to `nanos()` in any way but explicitly states it's getting some form of system timestamp...
On top of that you could rename `nanos` -> `monotonicNanos`, `millis` -> `monotonicMillis`?

Just wild ideas

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/Time.scala]: Millis were always used as a timestamp and nanos for timing, which is correct. The only thing is that there is no actual relation between nanoTime and current time.

Fixed it in the tests.

Comment (marek1840) [reply metals/src/main/scala/scala/meta/internal/metals/Time.scala]: To be honest, I don't see the benefit of using nanos here.
There is one place where nanos are used directly: to calculate the [StatusBar priority](https://github.com/scalameta/metals/blob/master/metals/src/main/scala/scala/meta/internal/metals/StatusBar.scala#L174) needed for sorting the statuses. I would argue that using milliseconds here should not be a noticeable change.

Therefore I would like to propose dropping the support for nanoseconds.

Comment (marek1840) [reply metals/src/main/scala/scala/meta/internal/metals/Time.scala]: Of course there is another solution - never ever using this method for creating timestamps, but then that is an additional complexity

File: metals/src/main/scala/scala/meta/internal/metals/Time.scala [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
  */
 trait Time {
   def nanos(): Long
-  final def millis(): Long = TimeUnit.NANOSECONDS.toMillis(nanos())
+  def millis(): Long
 }
 
 object Time {
   object system extends Time {
     def nanos(): Long = System.nanoTime()
```
Comment (marek1840) [inline metals/src/main/scala/scala/meta/internal/metals/Time.scala]: We can also include the "absolute time base" for proper calculation of the elapsed nanoseconds:
```suggestion
    private val absoluteTimeBaseNanos: Long = {
      val now = Instant.now()
      val seconds = TimeUnit.SECONDS.toNanos(now.getEpochSecond)
      val nanos = now.getNano
      seconds + nanos
    }

    private val relativeTimeBaseNanos = System.nanoTime()

    def nanos(): Long = {
      absoluteTimeBaseNanos + elapsed
    }

    private def elapsed: Long = System.nanoTime() - relativeTimeBaseNanos
```


Note, that this does not handle the time zones at all. Probably it is safe to assume the time of the server as the valid one. At least for now

File: metals/src/main/scala/scala/meta/internal/metals/Time.scala (hunk 1 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/Time.scala]: I think we can leave nanos for StatusBar priority. I will rename `millis` to `currentMillis` to indicate the difference.

File: metals/src/main/scala/scala/meta/internal/metals/Time.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/Time.scala]: I think it might be too much complexity for now.
